src/mol-plugin-state/manager/focus-camera/focus-first-residue.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2022 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Ke Ma <mark.ma@rcsb.org>
+ */
+import { Structure } from '../../../mol-model/structure';
+import { Vec3 } from '../../..//mol-math/linear-algebra/3d/vec3';
+import { PluginContext } from '../../../mol-plugin/context';
+import { CameraFocusOptions } from '../camera';
+import { PrincipalAxes } from '../../../mol-math/linear-algebra/matrix/principal-axes';
+import { Camera } from '../../../mol-canvas3d/camera';
+import { StructureComponentRef } from '../structure/hierarchy-state';
+
+export function getPolymerPositions(polymerStructure: Structure): Float32Array {
+    let positionIndex = 0;
+    const tmpMatrix = Vec3.zero();
+    const positions = new Float32Array(polymerStructure.atomicResidueCount * 3);
+    for (let i = 0; i < polymerStructure.units.length; i++) {
+        const unit = polymerStructure.units[i];
+        const { polymerElements } = unit.props;
+        if (polymerElements) {
+            for (let j = 0; j < polymerElements.length; j++) {
+                unit.conformation.position(polymerElements[j], tmpMatrix);
+                positions[positionIndex] = tmpMatrix[0];
+                positions[positionIndex + 1] = tmpMatrix[1];
+                positions[positionIndex + 2] = tmpMatrix[2];
+                positionIndex += 3;
+            }
```

Comment:
I think it might be better to use chunked array here to avoid edge cases:

```ts
const cAdd3 = ChunkedArray.add3;
const positions = ChunkedArray.create(Float32Array, 3, 1024, polymerStructure.atomicResidueCount);
cAdd3(positions, p[0], p[1], p[2]);
```

-----------

Also, it's better to cache `unit.conformation.position` outside of the loop for better performance.

Reply:
Sorry for late reply here. What do you mean, the chunked array till create a `Float32Array`. Just call `ChunkedArray.compact` to assign the final array.

Reply:
I just updated it 👍